cpp/src/Ice/Makefile.mk

```diff
@@ -16,7 +15,11 @@ Ice_extra_sources       := $(filter-out src/Ice/SSL/OpenSSL%.cpp src/Ice/SSL/Sch
 else
 Ice_extra_sources       := $(filter-out src/Ice/SSL/SecureTransport%.cpp src/Ice/SSL/Schannel%.cpp, $(wildcard src/Ice/SSL/*.cpp))
 endif
+
 Ice_excludes            = src/Ice/DLLMain.cpp
+Ice[shared]_excludes    = src/Ice/RegisterPluginsInit_min.cpp
```

Comment:
That's how I can use the same flags for static/shared build and yet get a small configuration-specific behavior: include a different impl file depending on the configuration.